articles/rules/redirect.md

```diff
@@ -29,27 +29,32 @@ function (user, context, callback) {
 }
 ```
 
-Once all rules have finished executing, the user will be redirected to the specified URL.
+Once all rules have finished executing, the user will be redirected to the specified URL. 
+
+Auth0 will also pass a state value in that URL, for example `https://example.com/foo?state=abc123`.
+
+<div class="alert alert-info"><strong>What is state?</strong> State is an opaque value the clients adds to the initial request that Auth0 includes when redirecting back to the client. We highly recommend using this param, since it can be used by the client to prevent <a href="/security/common-threats#cross-site-request-forgery-xsrf-or-csrf-">CSRF attacks</a>.</div>
```

Comment:
True in general, but in this case `state` is an opaque value that Auth0 passes to the redirect URL (which may be entirely independent of the client) - could you fix the wording?

If you look at my earlier review's explanation, this paragraph talks about `state` as being `foo123`, where we're actually documenting `bar42` (in that example). `bar42` was the state between Auth0 and the redirect URL domain - so it isn't including in the initial request to Auth0, nor is it sent back to the client.

It is correct that both states are there to prevent CSRF, though, so we can leave that part in.